### ISSUE TITLE
Top-p/Top-k Sampling Params handling in VLLM v1

### ISSUE THREAD

**Issue by terrykong** [NVIDIA-NeMo/RL#69]

**Comment by Dazz993** [comment]
Just for notes: there are some good discussions for Top-p/Top-k support in https://github.com/NVIDIA-NeMo/RL/pull/766.

---

raw logprobs are fine for now (same as v1). But are actually incorrect for top-k, top-p sampling (even though it _appears_ as though it would work correctly). Both the logprobs from vllm and the training framework are incorrect since they haven't been processed, which introduces error. (I believe the claim from #773 that v1 is fine is incorrect)

We have 2 options for how to mitigate this:
- use the post-processing from #773 on _both_ vllm raw_logprobs and megatron training
- use vllm post-processed logprobs and apply the processing to just megatron.

I'm leaning strongly towards the latter as it leaves less room for us to make mistakes and not notice. We should track this in #69.

_Originally posted by @SahilJain314 in https://github.com/NVIDIA-NeMo/RL/issues/766#issuecomment-3140991337_
            
---

> raw logprobs are fine for now (same as v1). But are actually incorrect for top-k, top-p sampling (even though it _appears_ as though it would work correctly). Both the logprobs from vllm and the training framework are incorrect since they haven't been processed, which introduces error. (I believe the claim from #773 that v1 is fine is incorrect)
> 
> We have 2 options for how to mitigate this:
> 
> * use the post-processing from [[1/2] Top-k and Top-p support for dtensor worker with vLLM V0 when `TP==1` #773](https://github.com/NVIDIA-NeMo/RL/pull/773) on _both_ vllm raw_logprobs and megatron training
> * use vllm post-processed logprobs and apply the processing to just megatron.
> 
> I'm leaning strongly towards the latter as it leaves less room for us to make mistakes and not notice. We should track this in #69.

I feel so too, although `raw_logprobs` is ok for temperature, the logits when using top-k/top-p will become incorrect if not actually applying top-k/top-p on them. And agreed with the latter one.

I think we can get this PR with `raw_logprobs` merged first. Then with @Dazz993 's help on https://github.com/NVIDIA-NeMo/RL/pull/773 and https://github.com/NVIDIA-NeMo/RL/pull/774, to support it at training side (both dtensor and megatron) and change `raw_logprobs` to `processed_logprobs` that time.

_Originally posted by @yuki-666 in https://github.com/NVIDIA-NeMo/RL/issues/766#issuecomment-3142564033_

**Comment by Dazz993** [comment]
Please see #773 for up-to-date discussions.

**Comment by yuki-97** [comment]
completed through https://github.com/NVIDIA-NeMo/RL/pull/2053
